waggle-dance-core/src/main/java/com/hotels/bdp/waggledance/client/tunneling/LocalPortFactory.java

```diff
@@ -0,0 +1,18 @@
+package com.hotels.bdp.waggledance.client.tunneling;
+
+import java.io.IOException;
+import java.net.ServerSocket;
+
+import com.hotels.hcommon.ssh.SshException;
+
+class LocalPortFactory {
```

Comment:
This looks a bit odd as a factory, couldn't it just be a method in the class where you use it from?

Reply:
I had it to mock the local port for TunnelingMetaStoreClientFactoryTest in `localHiveConfigUsesCorrectParameters`. But now I don't think there's much point in doing that because there is already an assertion for other parameters. So I've turned the LocalPortFactory in a private method.